### PR TITLE
bots: Move images directory and tools from test to bots

### PR DESCRIPTION
**Pull request by stefwalter**
The plan is that the test directory eventually operates on a fully formed test subject: The VM image to test with Cockpit already included.
    
It will be the job of code and data in the bots directory to prepare such an image.
    
Initially with this commit, we just move the image data to the bots directory so that it transcends can be used independent of the various branches. The naughty data corresponds to the images as well.

The bots remain common across various branches. They contain the logic, code and data for the bots to operate on the repository as a whole.
    
When rebasing during testing, if the branch does not contain a bots directory, checkout the bots from master.

Depends on:

 * [x] #6510 
 * [x] https://github.com/cockpit-project/cockpit/pull/6558
 * [x] #6576

### CODE REVIEW

File: bots/image-download [hunk 1]
```diff
@@ -42,12 +42,9 @@ import urlparse
 
 from contextlib import contextmanager
 
-BASE = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
-IMAGES = os.path.join(BASE, "images")
-DATA = os.path.join(os.environ.get("TEST_DATA", BASE), "images")
-
-# this is the path git uses within the repo, not on the filesystem (which would be IMAGES)
-GIT_IMAGE_PATH = "test/images"
+BOTS = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
```
Comment (dperpeet) [inline bots/image-download]: `os.path.dirname(os.path.realpath(__file__))` should be sufficient here
[realpath](https://docs.python.org/2/library/os.path.html#os.path.realpath) already returns the canical format

File: bots/image-download [hunk 2]
```diff
@@ -224,12 +222,14 @@ def prune_images(force, dryrun):
...
     # this hinges on being in the top level directory of the the git checkout
     with remember_cwd():
-        os.chdir(os.path.join(BASE, ".."))
+        os.chdir(os.path.join(BOTS, ".."))
         refs = get_refs()
         # list images present in each branch
         for name, ref in refs.items():
             sys.stderr.write("Considering images from {0} ({1})\n".format(name, ref))
-            for link in get_image_links(ref):
+            for link in get_image_links(ref, "test/images"):
```
Comment (dperpeet) [inline bots/image-download]: add a comment here that this is the deprecated, older format?
`# DEPRECATED old branches may still have images in test/images`

File: bots/image-prepare [hunk 3]
```diff
@@ -1,8 +1,7 @@
-# -*- coding: utf-8 -*-
-
+#!/usr/bin/env python
 # This file is part of Cockpit.
 #
-# Copyright (C) 2013 Red Hat, Inc.
+# Copyright (C) 2015 Red Hat, Inc.
```
Comment (dperpeet) [inline bots/image-prepare]: Is this from the history? Or what should we write here?

File: bots/image-prepare [hunk 4]
```diff
@@ -17,50 +16,122 @@
...
+    return 0
 
 def upload_scripts(machine, args):
     machine.execute("rm -rf /var/lib/testvm")
-    machine.upload([ os.path.join(testvm.TEST_DIR, "images", "scripts", "lib") ], "/var/lib/testvm")
-    machine.upload([ os.path.join(testvm.TEST_DIR, "images", "scripts", "%s.install" % machine.image) ], "/var/tmp")
-    machine.upload([os.path.join(testvm.TEST_DIR,"..", "containers")], "/var/tmp")
+    machine.upload([ os.path.join(BOTS, "images", "scripts", "lib") ], "/var/lib/testvm")
+    machine.upload([ os.path.join(BOTS, "images", "scripts", "%s.install" % machine.image) ], "/var/tmp")
+    machine.upload([os.path.join(BASE, "containers")], "/var/tmp")
```
Comment (dperpeet) [inline bots/image-prepare]: nitpick: if we're changing this line anyway, we might as well add spaces to be consistent with the previous lines

File: bots/image-download (hunk 1 above)
Comment (stefwalter) [inline bots/image-download]: Done.

File: bots/image-download (hunk 2 above)
Comment (stefwalter) [inline bots/image-download]: Done

File: bots/image-prepare (hunk 4 above)
Comment (stefwalter) [inline bots/image-prepare]: Done.

File: bots/image-prepare (hunk 3 above)
Comment (stefwalter) [inline bots/image-prepare]: Changed back.

File: bots/image-download [hunk 5]
```diff
@@ -209,10 +224,31 @@ def remember_cwd():
...
+    # iterate over visible refs (mostly branches)
+    # this hinges on being in the top level directory of the the git checkout
+    with remember_cwd():
+        os.chdir(os.path.join(BOTS, ".."))
+        refs = get_refs(open_pull_requests, offline)
+        # list images present in each branch / pull request
+        for name, ref in refs.items():
+            if not quiet:
+                sys.stderr.write("Considering images from {0} ({1})\n".format(name, ref))
+            for link in get_image_links(ref, "bots/images"):
```
Comment (dperpeet) [inline bots/image-download]: This needs to consider the old image links like https://github.com/cockpit-project/cockpit/pull/6511/files#diff-91824d32b0dcaf20aef463b2982b2b90R269

File: bots/image-download [hunk 6]
```diff
@@ -224,12 +260,16 @@ def prune_images(force, dryrun):
     # iterate over all visible refs (mostly branches)
     # this hinges on being in the top level directory of the the git checkout
     with remember_cwd():
-        os.chdir(os.path.join(BASE, ".."))
+        os.chdir(os.path.join(BOTS, ".."))
         refs = get_refs()
```
Comment (dperpeet) [inline bots/image-download]: `get_image_names` should already get all the images from the refs... maybe just add the images from the current files? They might not be checked in.

**Comment by dperpeet** [comment]
fixup proposal for pruning: https://github.com/dperpeet/cockpit/commit/a1212de7bddaa6c0c5b15d24587b0e959a560cbb

File: bots/image-download (hunk 5 above)
Comment (stefwalter) [inline bots/image-download]: Added.

File: bots/image-download (hunk 6 above)
Comment (stefwalter) [inline bots/image-download]: Not sure what this means. I'll assume it's just thinking out loud. The pruning needs to move to its own file/task anyway.

**Comment by stefwalter** [comment]
> fixup proposal for pruning: dperpeet/cockpit@a1212de

Added.

**Pull request merged**